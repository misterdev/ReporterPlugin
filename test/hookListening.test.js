const webpack = require('webpack');
const mockProcess = require('jest-mock-process');

const ReporterPlugin = require('../src/index');

const { mockReporter } = require('./helpers');

describe('ReporterPlugin', () => {
  it('should log the default hooks when using "defaults: true"', (done) => {
    const mockStdout = mockProcess.mockProcessStdout();
    const compiler = webpack({
      mode: 'none',
      context: __dirname,
      entry: './fixtures/a',
      plugins: [
        new ReporterPlugin({
          hooks: {
            defaults: true,
          },
          reporters: [mockReporter()],
        }),
      ],
    });
    compiler.run((err, stats) => {
      expect(mockStdout).toMatchSnapshot();
      mockStdout.mockRestore();
      done();
    });
  });

  it('should not print when setting "defaults: false"', (done) => {
    const mockStdout = mockProcess.mockProcessStdout();
    const compiler = webpack({
      mode: 'none',
      context: __dirname,
      entry: './fixtures/a',
      plugins: [
        new ReporterPlugin({
          hooks: {
            defaults: false,
          },
        }),
      ],
    });
    compiler.run((err, stats) => {
      expect(mockStdout).not.toHaveBeenCalled();
      mockStdout.mockRestore();
      done();
    });
  });

  it('should not listen default hooks when using "defaults: false"', (done) => {
    const mockStdout = mockProcess.mockProcessStdout();
    const compiler = webpack({
      mode: 'none',
      context: __dirname,
      entry: './fixtures/a',
      plugins: [
        new ReporterPlugin({
          hooks: {
            defaults: false,
            compiler: {
              done: true,
            },
          },
          reporters: [mockReporter()],
        }),
      ],
    });
    compiler.run((err, stats) => {
      expect(mockStdout).toHaveBeenCalledWith('Compilation finished\n');
      mockStdout.mockRestore();
      done();
    });
  });

  it('should throttle', (done) => {
    const mockStdout = mockProcess.mockProcessStdout();
    const compiler = webpack({
      mode: 'none',
      context: __dirname,
      entry: {
        a: './fixtures/a',
        b: './fixtures/b',
        c: './fixtures/c',
        d: './fixtures/d',
        e: './fixtures/e',
      },
      plugins: [
        new ReporterPlugin({
          hooks: {
            compilation: {
              buildModule: 2,
            },
          },
          reporters: [mockReporter()],
        }),
      ],
    });
    compiler.run((err, stats) => {
      expect(mockStdout).toMatchSnapshot();
      mockStdout.mockRestore();
      done();
    });
  });
});
