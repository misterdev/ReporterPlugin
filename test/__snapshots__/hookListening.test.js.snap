// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReporterPlugin should log the default hooks when using "defaults: true" 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Compilation starting...
",
    ],
    Array [
      "running
",
    ],
    Array [
      "before compiling
",
    ],
    Array [
      "compiling
",
    ],
    Array [
      "compilation
",
    ],
    Array [
      "compilation.buildModule
",
    ],
    Array [
      "finish module graph
",
    ],
    Array [
      "sealing
",
    ],
    Array [
      "dependencies optimization
",
    ],
    Array [
      "after dependencies optimization
",
    ],
    Array [
      "chunk graph
",
    ],
    Array [
      "after chunk graph
",
    ],
    Array [
      "optimizing
",
    ],
    Array [
      "module optimization
",
    ],
    Array [
      "after module optimization
",
    ],
    Array [
      "chunk optimization
",
    ],
    Array [
      "after chunk optimization
",
    ],
    Array [
      "module and chunk tree optimization
",
    ],
    Array [
      "after module and chunk tree optimization
",
    ],
    Array [
      "chunk modules optimization
",
    ],
    Array [
      "after chunk modules optimization
",
    ],
    Array [
      "module reviving
",
    ],
    Array [
      "before module ids
",
    ],
    Array [
      "module ids
",
    ],
    Array [
      "module id optimization
",
    ],
    Array [
      "module id optimization
",
    ],
    Array [
      "chunk reviving
",
    ],
    Array [
      "before chunk ids
",
    ],
    Array [
      "chunk id optimization
",
    ],
    Array [
      "after chunk id optimization
",
    ],
    Array [
      "record modules
",
    ],
    Array [
      "record chunks
",
    ],
    Array [
      "hashing
",
    ],
    Array [
      "content hashing
",
    ],
    Array [
      "after hashing
",
    ],
    Array [
      "record hash
",
    ],
    Array [
      "module assets processing
",
    ],
    Array [
      "chunk assets processing
",
    ],
    Array [
      "additional chunk assets processing
",
    ],
    Array [
      "recording
",
    ],
    Array [
      "additional asset processing
",
    ],
    Array [
      "chunk asset optimization
",
    ],
    Array [
      "after chunk asset optimization
",
    ],
    Array [
      "asset optimization
",
    ],
    Array [
      "after asset optimization
",
    ],
    Array [
      "after seal
",
    ],
    Array [
      "emit
",
    ],
    Array [
      "Compilation finished
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": true,
    },
  ],
}
`;
